src/pybind/mgr/cephadm/module.py

```diff
@@ -2360,26 +2371,38 @@ def _calc_daemon_deps(self,
             root_cert = ''
             server_port = ''
             try:
-                server_port = str(self.cherrypy_thread.server_port)
-                root_cert = self.cherrypy_thread.ssl_certs.get_root_cert()
+                server_port = str(self.http_server.agent.server_port)
+                root_cert = self.http_server.agent.ssl_certs.get_root_cert()
             except Exception:
                 pass
             deps = sorted([self.get_mgr_ip(), server_port, root_cert,
                            str(self.device_enhanced_scan)])
         elif daemon_type == 'iscsi':
             deps = [self.get_mgr_ip()]
+
+        elif daemon_type == 'prometheus':
+            # for prometheus we add the active mgr as an explicit dependency,
+            # this way we force a redeploy after a mgr failover
+            deps.append(self.get_active_mgr().name())
+            deps.append(str(self.get_module_option_ex('prometheus', 'server_port', 9283)))
+            node_exporter_cnt = len(self.cache.get_daemons_by_service('node-exporter'))
+            alertmgr_cnt = len(self.cache.get_daemons_by_service('alertmanager'))
+            haproxy_cnt = len(self.cache.get_daemons_by_type('ingress'))
+            if alertmgr_cnt > 0:
+                deps.append('alertmanager')
+            if haproxy_cnt > 0:
+                deps.append('ingress')
+            if node_exporter_cnt > 0:
+                deps.append('node-exporter')
```

Comment:
Don't repeat:
```suggestion
            for service in ['node-exporter', 'alertmanager', 'ingress']:
                if self.cache.get_daemons_by_service(service):
                    deps.append(service)
```
Or even better:
```suggestion
            deps += [
                svc for svc in ['node-exporter', 'alertmanager', 'ingress'] if self.cache.get_daemons_by_service(svc)]
```

Reply:
Fixed by using the shortest version :+1: